public/editor/scripts/editor/js/fc/bramble-ui-bridge.js

```diff
@@ -11,10 +11,72 @@ define(function(require) {
 
   var _escKeyHandler;
 
+  var adapting = false;
+  var adaptTimeoutMS = 200; // How often we adapt editor bar layout
+
   function updateLayout(data) {
     $(".filetree-pane-nav").width(data.sidebarWidth);
     $(".editor-pane-nav").width(data.firstPaneWidth);
     $(".preview-pane-nav").width(data.secondPaneWidth);
+
+    // Only adapt the layout every once in a while
+    if(!adapting) {
+      adapting = true;
+      adaptLayout();
+      setTimeout(function(){
+        adapting = false;
+      },adaptTimeoutMS);
+    }
+  }
+
+  // Adapt each of the pane header elements
+  function adaptLayout(){
+    $(".nav-container").each(function(){
+      adaptElement($(this));
+    });
+  }
+
+  // Checks if there is enough room for all of the elements inside it
+  // Adds a 'narrow' class, in priority order, when there isn't.
+  function adaptElement(el){
+    var itemCount = el.find("[adapt-order]").length;
+    el.find("[adapt-order]").addClass("narrow");
+
+    for(var i = itemCount; i > 0; i--) {
+      if(hasEnoughRoom(el)) {
+        el.find("[adapt-order="+i+"]").removeClass("narrow");
```

Comment:
This would remove the narrow class form all adapt-elements, showing them all, but it's just supposed to show the one it's currently trying to fit. Am I misreading?